spikeinterface/curation/auto_merge.py

```diff
@@ -0,0 +1,264 @@
+import numpy as np
+import scipy.signal
+import scipy.spatial
+
+from ..postprocessing import compute_correlograms, get_template_extremum_channel
+
+# TODO:
+#   * adaptative window p(aka plage) on CC
+#   * template similarity
+#   * 
+
+def get_potential_auto_merge(waveform_extractor,
+                minimum_spikes=1000, maximum_distance_um=200.,
+                peak_sign="neg",
+                bin_ms=0.25, window_ms=50., corr_thresh=0.3,
+                correlogram_low_pass = 800., adaptative_window_threshold=0.5,
+                debug_folder=None,
+                ):
+    """
+    Algorithm to find and check potential merges between units.
+    
+    This is taken from lussac version 1 done Aurelien Wyngaard.
+    https://github.com/BarbourLab/lussac/blob/main/postprocessing/merge_units.py
+    
+    This check:
+       * correlograms
+       * template_similarity
+       * contamination quality mertics
+    
+    """
+    
+    we = waveform_extractor
+    sorting = we.sorting
+    unit_ids = sorting.unit_ids
+    
+    # STEP 1 : to get fast computation we will not analyse pairs when:
+    #    * not enough spikes for one of theses
+    #    * to far away one from each other
+    n = unit_ids.size
+    pair_mask = np.ones((n, n), dtype='bool')
```

Comment:
Steps 1 and 2 remove units and not pairs.
Can't we detect the units to remove and create `UnitsSelectionSorting` (or equivalent for the WaveformExtractor).
This could save quite a bit of time for future calculations, and then the mask would exist only for the pair of remaining units.

Reply:
In the actual code, the step1 and step2 remove the pair to be explored and so computed.
I think this is more flexible and the final results is the same for the same speed.

With this approcach, so we can easily switch steps.
Morover, this function could be call with a pre-cleaned or not units list.

Reply:
But for the correlograms computation, the speed goes with the number of units **squared**. So going from something like 100 to 60 units could make the computation ~2.8x faster.
If you have a long recording with high firing-rate units, I think it can make a difference.